cache/manager.go

```diff
@@ -709,7 +710,7 @@ func (cm *cacheManager) prune(ctx context.Context, ch chan client.UsageInfo, opt
 
 			c := &client.UsageInfo{
 				ID:         cr.ID(),
-				Mutable:    cr.mutable,
+				Mutable:    !cr.isFinalized,
```

Comment:
Maybe this should be `cr.mutableRef != nil` which indicates this isn't committed (rather than finalized)?

Reply:
So there's a pre-existing naming issue that makes this all pretty confusing: when `mutableRef.Commit` is called, it's not actually committed, its snapshot is left as mutable and can later be retrieved as mutable via `cacheManager.GetMutable`. It's only once `finalize` has been called on a ref that it's actually committed.

So I set this as `!isFinalized` because until the ref is finalized, the underlying snapshot is still technically mutable even if it's also possible to access it via an `ImmutableRef`. I don't think we should use `cr.mutableRef != nil` because `cr.mutableRef` just indicates whether there is an open mutable ref to the cache record, but a mutable cache record can exist without any open refs to it (if cache policy is set to retain).

AFAIK this `Mutable` field in `UsageInfo` is only used for informational purposes, so I think this should all be a safe change from before (@tonistiigi can confirm).

In terms of the confusing naming, I already have a bunch more PRs queued up refactoring stuff in the cache package, I think I will include another one that changes the naming here so that
1. `MutableRef.Commit()` is renamed to `MutableRef.ToImmutable()`, which better reflects what it actually does
2. `cacheRecord.isFinalized` can then be renamed to `cacheRecord.isCommitted` without confusion

Just don't want to add more interface changes to this PR as that's a separate issue.

Reply:
Thank you for explaining this. Let's leave it as `!cr.isFinalized` that indicates the mutability of the record